lib/driver.js

```diff
@@ -2,18 +2,38 @@ import { MobileJsonWireProtocol,
          errors } from 'mobile-json-wire-protocol';
 import commands from './commands';
 import log from './logger';
+import { desiredCapabilityConstraints } from './desired-caps';
 import B from 'bluebird';
 import _ from 'lodash';
+import validator from 'validate.js';
+
+validator.promise = B;
+validator.prettify = (val) => { return val; };
+
+const NEW_COMMAND_TIMEOUT_MS = 60 * 1000;
 
 class BaseDriver extends MobileJsonWireProtocol {
 
-  constructor () {
+  constructor (shouldValidateCaps = true) {
     super();
     this.sessionId = null;
     this.caps = null;
+    this.newCommandTimeoutMs = NEW_COMMAND_TIMEOUT_MS;
+    this._constraints = _.cloneDeep(desiredCapabilityConstraints);
+
+    // base-driver internals
     this.curCommand = new B((r) => { r(); }); // see note in execute
     this.noCommandTimer = null;
-    this.newCommandTimeoutMs = null;
+    this.shouldValidateCaps = shouldValidateCaps;
+  }
+
+  // we only want subclasses to ever extend the contraints
+  set desiredCapConstraints (constraints) {
+    this._constraints = _.merge(this._constraints, constraints);
```

Comment:
originally I wasn't cloning the default contraits, so object.assign was mutating the default for all further drivers. Now that I'm cloning, object.assign works again :)